Main.java

```diff
@@ -2,37 +2,49 @@
-        int  shamanDmg = sc.nextInt();
-        Unit warrior = new Unit(warriorDmg, warriorHp);
-        Unit shaman = new Unit(shamanDmg, shamanHp);
-        System.out.println("Warrior élete és sebzése: "+ warrior.hp +", "+ warrior.dmg + " || Shaman élete és sebzése: " + shaman.hp + ", " + shaman.dmg + '\n');
-        battle(warrior,shaman);
-        sc.close();
-    } 
+        initialize();
+    }
 
-    private static void battle (Unit our, Unit enemy){
+    private static void battle(Unit unit1, Unit unit2) {
+        System.out.println("A Csata elkezdődött! " + unit1.name + " kezdheti a támadást " + unit2.name + (" ellen."));
         int i = 1;
-        while(our.isAlive() && enemy.isAlive()){
-            our.attack(enemy);
-            if(our.isAlive() && enemy.isAlive()){
-                enemy.attack(our);
-                if (our.hp < 0 || enemy.hp < 0)
-                    System.out.println(i + ". körben a maradék életünk: " + our.hp + ", lenne az ellenség maradék élete: " + enemy.hp + " lenne");
-                else
-                    System.out.println(i + ". körben a maradék életünk: " + our.hp + ", az ellenség maradék élete: " + enemy.hp);
-                i++;
+        while (unit1.isAlive() && unit2.isAlive()) {
+            unit1.attack(unit2);
+            System.out.println(i + ". körben " + unit1.name + " támad --- " + unit1.name + " élete: " + unit1.hp
+                    + " || " + unit2.name + " élete: " + unit2.hp);
+            if (!unit2.isAlive()) {
+                break;
             }
+            unit2.attack(unit1);
+            i++;
+            System.out.println(i + ". körben " + unit2.name + " támad, ---" + unit1.name + " élete: " + unit1.hp
+                    + " || " + unit2.name + " élete: " + unit2.hp);
+
         }
+        System.out.println(unit1.isAlive() ? "\n" + unit1.name.toUpperCase() + " GYŐZEDELMESKEDETT!"
+                : "\n" + unit2.name.toUpperCase() + " GYŐZEDELMESKEDETT!");
+    }
 
-        System.out.println((our.hp>enemy.hp) ? "\nNyertünk!" : "\nVesztettünk!");
+    private static void initialize() {
+        Scanner sc = new Scanner(System.in);
+        // Első Unit
+        System.out.println("Első egység neve: ");
+        String unit1Name = sc.next();
+        System.out.println(unit1Name + " élete:");
+        int unit1Hp = sc.nextInt();
+        System.out.println(unit1Name + " sebzése: ");
+        int unit1Dmg = sc.nextInt();
+        // Második Unit
+        System.out.println("Második egység neve: ");
+        String unit2Name = sc.next();
+        System.out.println(unit2Name + " élete:");
+        int unit2Hp = sc.nextInt();
+        System.out.println(unit2Name + " sebzése: ");
+        int unit2Dmg = sc.nextInt();
+        // Két Unit létrehozása
+        Unit unit1 = new Unit(unit1Name, unit1Dmg, unit1Hp);
+        Unit unit2 = new Unit(unit2Name, unit2Dmg, unit2Hp);
+        sc.close();
+        battle(unit1, unit2);
```

Comment:
a fuggveny nevvalasztasa innentol kezdve annyira nem szerenses, mert ez nem csak inicializal, hanem mar a csatat is levezenyli.